src/fetchApi.js

```diff
@@ -14,7 +14,11 @@ export async function fetchApi(request) {
   })
     .then(async (response) => {
       const { headers, status } = response;
-      const responseBody = await response.json();
+      let responseBody;
+      const contentType = response.headers.get('content-type').toLowerCase();
+      if (contentType && contentType.includes('application/json')) {
+        responseBody = await response.json();
+      }
       return { request, status, headers, body: responseBody };
```

Comment:
so in the case of non-JSON, the body is just empty... Is that what we want or should we parse it raw?

Reply:
Can we parse it if it's empty?

What should we parse it with, https://developer.mozilla.org/en-US/docs/Web/API/Body/blob ?

Reply:
Oh I understand now.
Will the other parses not throw if the body is empty as well?

Reply:
No, they shouldn't throw. And yeah, probably as a blob